src/main/java/io/orangebeard/listener/OrangebeardExtension.java

```diff
@@ -74,29 +80,68 @@ public OrangebeardExtension() {
+        // If the element is already in the tree, then a test suite was already started for this, and we don't have to do anything.
+        // For every element NOT already in the tree, then we must start a new test suite, and add a node to the tree.
+        TestSuiteTree parentNode = root;
+        for (int i = 0; i < classNameComponents.length; i++) {
+            Optional<TestSuiteTree> currentNode = parentNode.getChildByName(classNameComponents[i]);
+            if (currentNode.isEmpty()) {
+                // Create and start the test suite.
+                StartTestItem startTestItem = new StartTestItem(testrunUUID, classNameComponents[i], SUITE, null, null);
+
+                // Get the UUID of the test suite in the parent node; use "null" if the parent node is the root node.
+                // The node key is usually the String representation of that UUID, but that is not guaranteed.
+                // This is why we keep track of a node's test suite UUID separately.
+                UUID uuidOfParent = null;
+                if (parentNode != root) {
+                    uuidOfParent = parentNode.getTestSuiteUUID();
+                }
+
+                // Now that we now the UUID of the parent suite, we can start a new suite as its child.
+                UUID suiteId = orangebeardClient.startTestItem(uuidOfParent, startTestItem);
+
+                // Add a node to the tree for this newly created and started test suite.
+                String key = suiteId.toString();
+                if (i == classNameComponents.length - 1) {
+                    key = extensionContext.getUniqueId();
+                }
+                currentNode = Optional.of(parentNode.addChild(classNameComponents[i], key, suiteId));
+            }
+            // Continue with the next level of the package hierarchy.
+            // At this point, `currentNode` is always filled.
+            parentNode = currentNode.get();
+        }
     }
 
     @Override
     public void afterAll(ExtensionContext extensionContext) {
-        UUID suiteId = suites.get(extensionContext.getUniqueId()).getUuid();
-
-        FinishTestItem finishTestItem = new FinishTestItem(testrunUUID, PASSED, null, null);
-        orangebeardClient.finishTestItem(suiteId, finishTestItem);
-        suites.remove(extensionContext.getUniqueId());
+        // The test suites that were started, must now be finished cleanly.
+        // After finishing a test suite, it should be removed from the tree.
+        // We only remove leaf nodes, because the intermediate nodes may be needed in the next suite of this test run.
+        List<TestSuiteTree> leaves = root.getLeaves();
+        for (TestSuiteTree leaf : leaves) {
+            UUID suiteId = leaf.getTestSuiteUUID();
+            FinishTestItem finishTestItem = new FinishTestItem(testrunUUID, PASSED, null, null);
+            orangebeardClient.finishTestItem(suiteId, finishTestItem);
+            leaf.detach();
```

Comment:
What's the point of doing this? after this method is finished the test run will finish and the class will be destroyed.